torch/_dynamo/optimizations/training.py

```diff
@@ -22,6 +31,62 @@
 log = logging.getLogger(__name__)
 
 
+def aot_autograd(**kwargs):
+    def compiler_fn(gm: torch.fx.GraphModule, example_inputs):
+        import functorch.compile
+
+        # Hack to get around circular import problems with aot_inductor_debug
+        if callable(kwargs.get("decompositions")):
+            kwargs["decompositions"] = kwargs["decompositions"]()
+
+        # TODO: stop monkeypatching here (without even cleaning up, UGH!)
+        functorch.compile.config.use_functionalize = True
+        functorch.compile.config.use_fake_tensor = True
+
+        force_compile_tiny_graphs = kwargs.pop("force_compile_tiny_graphs", False)
+
+        if count_calls(gm.graph) < 2 and not force_compile_tiny_graphs:
```

Comment:
as per the comment below for decomps, should we always force compilation?

Reply:
hmm yeah. I like the idea to force recompilation in debug mode anyway.

Reply:
@Chillee you cool with this idea?

Reply:
Yeah I'm fine.

Reply:
is it truly pointless? what if the single op decomposes into some ops that can be fused?

Reply:
we gonna remove this conditional!